objcgen/objcprocessor.cs

```diff
@@ -143,6 +144,13 @@ protected IEnumerable<MethodInfo> GetMethods (Type t)
 					continue;
 				}
 
+				if (implement_system_iequatable_t) {
+					if (mi.Match ("System.Boolean", "Equals", new string [] { null }) && !mi.Match ("System.Boolean", "Equals", "System.Object")) {
```

Comment:
Actually, shouldn't even need it as an Else, as long as it's checked after the Equals (since if it hits the if block for equals, it then calls 'continue'